src/plugins/dashboard/public/dashboard_top_nav/internal_dashboard_top_nav.tsx

```diff
@@ -313,10 +321,51 @@ export function InternalDashboardTopNav({
       });
     }
     if (showWriteControls && managed) {
-      allBadges.push(getManagedContentBadge(dashboardManagedBadge.getTooltip()));
+      const badgeProps = {
+        ...getManagedContentBadge(dashboardManagedBadge.getManagedText(), false),
+        onClickAriaLabel: dashboardManagedBadge.getAriaLabel(),
+        iconOnClick: () => setIsPopoverOpen(!isPopoverOpen),
+        iconOnClickAriaLabel: dashboardManagedBadge.getAriaLabel(),
+        onClick: () => setIsPopoverOpen(!isPopoverOpen),
+      } as TopNavMenuBadgeProps;
```

Comment:
**very very small nit:** I like to try to group things by something they have in common - for example:

```suggestion
        ...getManagedContentBadge(dashboardManagedBadge.getManagedText(), false),
        onClick: () => setIsPopoverOpen(!isPopoverOpen),
        onClickAriaLabel: dashboardManagedBadge.getAriaLabel(),
        iconOnClick: () => setIsPopoverOpen(!isPopoverOpen),
        iconOnClickAriaLabel: dashboardManagedBadge.getAriaLabel(),
      } as TopNavMenuBadgeProps;
```

or...

```suggestion
        ...getManagedContentBadge(dashboardManagedBadge.getManagedText(), false),
        onClick: () => setIsPopoverOpen(!isPopoverOpen),
        iconOnClick: () => setIsPopoverOpen(!isPopoverOpen),
        onClickAriaLabel: dashboardManagedBadge.getAriaLabel(),
        iconOnClickAriaLabel: dashboardManagedBadge.getAriaLabel(),
      } as TopNavMenuBadgeProps;
```